src/app/state/importaccounts/saga.ts

```diff
@@ -71,33 +78,64 @@ function* enumerateAccountsFromMnemonic(action: PayloadAction<string>) {
   }
 }
 
+function* fetchBalanceForAccount(account: ImportAccountsListAccount) {
+  let currentStep = (yield* select(selectImportAccounts)).step
+  if (currentStep === ImportAccountsStep.Idle) {
+    yield* setStep(ImportAccountsStep.LoadingBalances)
+  }
+  const balance = yield* call(getBalance, hex2uint(account.publicKey))
+  yield* put(
+    importAccountsActions.updateAccountBalance({
+      address: account.address,
+      balance,
+    }),
+  )
+  currentStep = (yield* select(selectImportAccounts)).step
+  const hasMissingBalances = yield* select(selectImportAccountHasMissingBalances)
+  if (currentStep === ImportAccountsStep.LoadingBalances && !hasMissingBalances) {
+    yield* setStep(ImportAccountsStep.Idle)
+  }
+}
+
+function* ensureAllBalancesArePresentOnCurrentPage() {
+  const existingAccounts = yield* select(selectImportAccountsOnCurrentPage)
+  const accountsWithNoBalance = existingAccounts.filter(a => !a.balance)
+  for (let i = 0; i < accountsWithNoBalance.length; i++) {
+    yield* fork(fetchBalanceForAccount, accountsWithNoBalance[i])
+  }
+}
+
+/**
+ * Enumerate more accounts from Ledger, enough to fill up one page.
+ */
 function* enumerateAccountsFromLedger() {
+  const existingAccounts = yield* select(selectImportAccountsFullList)
+  const pageNumber = yield* select(selectImportAccountsPageNumber)
+  if (existingAccounts.length >= (pageNumber + 1) * accountsPerPage) return
   yield* setStep(ImportAccountsStep.OpeningUSB)
   let transport: Transport | undefined
   try {
     transport = yield* getUSBTransport()
+    const existingAccounts = yield* select(selectImportAccountsFullList)
+    const start = existingAccounts.length
 
     yield* setStep(ImportAccountsStep.LoadingAccounts)
-    const accounts = yield* call(Ledger.enumerateAccounts, transport)
+    for (let index = start; index < start + accountsPerPage; index++) {
+      const a = (yield* call(Ledger.enumerateAccounts, transport, 1, index))[0]
```

Comment:
let's use meaningful var names

Reply:
Originally I used account, but then I wanted to minimize my changes (compared to the original code) and when back to "a". Going back to "account" now.